toolchain/sem_ir/ids.h

```diff
@@ -735,6 +735,11 @@ struct TypeId : public IdBase<TypeId> {
   // Returns the constant ID that defines the type.
   auto AsConstantId() const -> ConstantId { return ConstantId(index); }
 
+  // Returns whether this represents a symbolic type. Requires has_value.
+  auto is_symbolic() const -> bool { return AsConstantId().is_symbolic(); }
+  // Returns whether this represents a concrete type. Requires has_value.
+  auto is_concrete() const -> bool { return AsConstantId().is_concrete(); }
```

Comment:
It feels weird to add a convenience accessor that's unused... Per discussion, maybe replace `is_concrete` with `!is_symbolic`? It's weird because we also have a lot of `!is_concrete` uses which could be `is_symbolic`.